website/docs/r/upgrade_run.html.markdown

```diff
@@ -0,0 +1,105 @@
+Please use data source `nsxt_upgrade_postcheck` to retrieve results of upgrade post-checks.
+
+## Example Usage
+
+```hcl
+resource "nsxt_upgrade_run" "run1" {
+  upgrade_prepare_id = nsxt_upgrade_prepare.test.id
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg1.id
+    enabled = false
+  }
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg2.id
+    enabled = true
+  }
+
+  host_group {
+    id       = data.nsxt_edge_upgrade_group.eg2.id
+    parallel = true
+  }
+
+  edge_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `upgrade_prepare_ready_id` - (Required) ID of corresponding `nsxt_upgrade_prepare_ready` resource. Update this field will trigger replacement (destroy and create) of this resource.
+* `edge_group` - (Optional) EDGE component upgrade unit group configurations. Groups will be reordered following the order they present in this field.
+    * `id` - (Required) ID of the upgrade unit group.
+    * `enabled` - (Optional) Flag to indicate whether upgrade of this group is enabled or not. Default: True.
+    * `parallel` - (Optional) Upgrade method to specify whether the upgrade is to be performed in parallel or serially. Default: True.
+    * `pause_after_each_upgrade_unit` - (Optional) Flag to indicate whether upgrade should be paused after upgrade of each upgrade-unit. Default: False.
+* `host_group` - (Optional) HOST component upgrade unit group configurations. Groups will be reordered following the order they present in this field.
+    * `id` - (Required) ID of the upgrade unit group.
+    * `enabled` - (Optional) Flag to indicate whether upgrade of this group is enabled or not. Default: True.
+    * `parallel` - (Optional) Upgrade method to specify whether the upgrade is to be performed in parallel or serially. Default: True.
+    * `pause_after_each_upgrade_unit` - (Optional) Flag to indicate whether upgrade should be paused after upgrade of each upgrade-unit. Default: False.
+    * `upgrade_mode` - (Optional) Upgrade mode. Supported values: `maintenance_mode`, `in_place`, `stage_in_vlcm`.
+    * `maintenance_mode_config_vsan_mode` - (Optional) Maintenance mode config of vsan mode. Supported values: `evacuate_all_data`, `ensure_object_accessibility`, `no_action`.
+    * `maintenance_mode_config_evacuate_powered_off_vms` - (Optional) Maintenance mode config of whether evacuate powered off vms. Supported values: `true`, `false`.
+    * `rebootless_upgrade` - (Optional) Flag to indicate whether to use rebootless upgrade. Default: True.
+* `edge_upgrade_setting` - (Optional) EDGE component upgrade plan setting.
+    * `parallel` - (Optional) Upgrade Method to specify whether the upgrade is to be performed serially or in parallel. Default: True.
```

Comment:
I think we might want to clearly distinguish the upgrade semantics of this parallel setting vs the one in edge_group. This comment applies to host_upgrade_setting as well.

Reply:
Done.